### ISSUE TITLE
Validate invalid methods syntax

### ISSUE THREAD

**Issue by hyyan** [BBx-Kitchen/bbj-language-server#82]
The following syntax is invalid in BBj and will throw an error, the method syntax should be validated:
```
class public Test
  method private run()
  methodend
classend

t! = new Test()
t!.run()
```
```

```

**Comment by dhuebner** [comment]
@hyyan 
You mean missing return type, or what is invalid?

**Comment by hyyan** [comment]
@dhuebner Yes. The retune type is missing. The only place where you can omit the return type is in constructor.

**Comment by dhuebner** [comment]
@hyyan 
The grammar should be more lex in sense of the syntax where it is possible so the use can get better feedback, that why the type here is grammatically  optional.  We need a validation for that.
Thanks for reporting!

**Comment by StephanWald** [comment]
**Triage → actionable** (good first issue). Per this thread, the method return type is intentionally optional in the grammar (for recovery) and a validation is expected. Add a `MethodDecl` check: error when `!voidReturn && !returnType` and the method is **not** a constructor (name != enclosing class name); should also apply to interface methods. Home: `check-classes.ts`. Labeled `actionable`, `good first issue`.

**Comment by StephanWald** [comment]
**Already handled on `main` → closing as OBE.**

Verified against the current `main`: the example already produces an error. `t!.run()` on the private method reports:

```
The member 'run' from the type 'Test' (in <file>:3) is not visible
```

This is the existing private-member access check (see the `#198` visibility tests in `validation.test.ts`, e.g. `t!.doPrivately()` → "is not visible"). A method *declaration* without a return type is valid BBj (constructors, and methods like `method public doTest()`, are used throughout the test suite), so there is no separate "invalid method syntax" to flag here — the only invalid aspect of the sample is the external call to a private method, which is already diagnosed.

Closing as resolved. If you intended a different invalid-syntax case, please reopen with a specific example.
